Neos.Flow/Classes/ResourceManagement/Storage/FileSystemStorage.php

```diff
@@ -161,14 +158,6 @@ public function getObjectsByCollection(CollectionInterface $collection, callable
                 return $this->getStreamByResource($resource);
             });
             yield $object;
-            if ($callback !== null) {
-                call_user_func($callback, $iteration, $object);
-            }
-            $iteration++;
-        }
-        if ($iteration === 0) {
-            // required because if the collection is empty, this should still return an empty generator
-            yield from [];
```

Comment:
i doubt that this https://github.com/neos/flow-development-collection/pull/2167 fixed this. Should not be necessary but ill write a test ^^

Reply:
see added test ;)